### PR TITLE
obs_subaru/DM-7884: unification of coadd and diff datasets

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: policy/SuprimecamMapper.paf [hunk 1]
```diff
@@ -750,11 +701,9 @@ datasets: {
     # Measurements on coadds
 
     # Reference catalog for coadds
+    # dataset defined in obs_base modified in this mapper as follows:
```
Comment (TallJimbo) [inline policy/SuprimecamMapper.paf]: This `python` entry is definitely incorrect: it refers to the base class of the config class we want.  I suspect `obs_base` has the right one.

File: policy/SuprimecamMapper.paf [hunk 2]
```diff
@@ -572,17 +527,13 @@ datasets: {
...
         tables:        raw_visit
     }
+    # dataset defined in obs_base modified in this mapper as follows:
     forced_src_schema: {
         template:      "schema/forced_src.fits"
-        python:        "lsst.afw.table.SourceCatalog"
-        persistable:        "ignored"
-        storage:    "FitsCatalogStorage"
     }
+    # dataset defined in obs_base modified in this mapper as follows:
```
Comment (TallJimbo) [inline policy/SuprimecamMapper.paf]: This `python` entry is incorrect.  It should point to a class in `meas.base`, not `pipe.tasks`.  Hopefully that change will bring it into agreement with `obs_base`.

File: policy/HscMapper.paf [hunk 3]
```diff
@@ -193,14 +169,6 @@ exposures: {
...
     }
 
-    diff: {
-        template:      "diff/%(diff)05d/DIFF%(patch)08d.fits"
-        python:        "lsst.afw.image.ExposureF"
-        persistable:        "ExposureF"
-        storage:    "FitsStorage"
-        level:        "Skytile"
-        tables:        "raw_skytile"
-    }
```
Comment (PaulPrice) [inline policy/HscMapper.paf]: Please don't leave two blank lines.

File: policy/HscMapper.paf [hunk 4]
```diff
@@ -450,28 +418,7 @@ datasets: {
...
 
-    diffpsf: {
-        template:      "diff/%(diff)05d/PSF%(patch)08d%(filter)s.boost"
-        python:        "lsst.afw.detection.Psf"
-        persistable:   "Psf"
-        storage:       "BoostStorage"
-        tables:        "raw_skytile"
-    }
```
Comment (PaulPrice) [inline policy/HscMapper.paf]: Ditto.

File: policy/SuprimecamMapper.paf [hunk 5]
```diff
@@ -154,23 +146,7 @@ exposures: {
...
-    }
 
-    diff: {
-        template:      "diff/%(diff)05d/DIFF%(patch)08d.fits"
-        python:        "lsst.afw.image.ExposureF"
-        persistable:        "ExposureF"
-        storage:    "FitsStorage"
-        level:        "Skytile"
-        tables:        "raw_skytile"
-    }
```
Comment (PaulPrice) [inline policy/SuprimecamMapper.paf]: Ditto.

File: policy/SuprimecamMapper.paf [hunk 6]
```diff
@@ -402,28 +378,7 @@ datasets: {
...
 
-    diffpsf: {
-        template:      "diff/%(diff)05d/PSF%(patch)08d%(filter)s.boost"
-        python:        "lsst.afw.detection.Psf"
-        persistable:   "Psf"
-        storage:       "BoostStorage"
-        tables:        "raw_skytile"
-    }
```
Comment (PaulPrice) [inline policy/SuprimecamMapper.paf]: Ditto.

File: policy/HscMapper.paf [hunk 7]
```diff
@@ -514,11 +514,9 @@ datasets: {
         tables:        "raw"
         tables:        "raw_visit"
     }
+    # dataset defined in obs_base modified in this mapper as follows:
     deepCoadd_calexp_background: {
         template:      "deepCoadd-results/%(filter)s/%(tract)d/%(patch)s/det_bkgd-%(filter)s-%(tract)d-%(patch)s.fits"
```
Comment (PaulPrice) [inline policy/HscMapper.paf]: Why is the HSC `template` different?  Is there a reason not to just adopt the obs_base one?

File: policy/SuprimecamMapper.paf [hunk 8]
```diff
@@ -527,17 +527,13 @@ datasets: {
         tables:        raw
         tables:        raw_visit
     }
+    # dataset defined in obs_base modified in this mapper as follows:
     forced_src_schema: {
         template:      "schema/forced_src.fits"
```
Comment (PaulPrice) [inline policy/SuprimecamMapper.paf]: Again, I think we should use what's in obs_base unless there's a good reason not to.

File: policy/SuprimecamMapper.paf (hunk 2 above)
Comment (pgee2000) [inline policy/SuprimecamMapper.paf]: I have this fix listed on RFC-237.  Will do it next week.

File: policy/SuprimecamMapper.paf (hunk 1 above)
Comment (pgee2000) [inline policy/SuprimecamMapper.paf]: I have this fix listed on RFC-237.  Will do it next week.

**Pull request merged**